File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/domain/FasePresenca.java
package br.ufpr.sept.so2.modules.presenca.domain;

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException;

public enum FasePresenca {
    ENTRADA,
    SAIDA;

    public static FasePresenca from(String raw) {
        if (raw == null || raw.isBlank()) {
            throw new DadoInvalidoException("Fase de presença obrigatória.");
        }
        try {
            return FasePresenca.valueOf(raw.trim().toUpperCase());
        } catch (IllegalArgumentException ex) {
            throw new DadoInvalidoException("Fase de presença inválida.");
        }
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/domain/FasePresenca.kt
package br.ufpr.sept.so2.modules.presenca.domain

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException

enum class FasePresenca {
    ENTRADA,
    SAIDA,
    ;

    companion object {
        fun from(raw: String?): FasePresenca {
            if (raw.isNullOrBlank()) {
                throw DadoInvalidoException("Fase de presença obrigatória.")
            }
            return try {
                valueOf(raw.trim().uppercase())
            } catch (_: IllegalArgumentException) {
                throw DadoInvalidoException("Fase de presença inválida.")
            }
        }
    }
}
